src/main/resources/assets/paths/datasetV3ById.yaml

```diff
@@ -0,0 +1,82 @@
+put:
+  summary: Updates a dataset from a valid DatasetUpdate object
+  description: Updates a dataset from a valid DatasetUpdate object, returns a full dataset object
+  parameters:
+    - name: datasetId
+      in: path
+      description: The Dataset ID
+      required: true
+      schema:
+        type: integer
+  requestBody:
+    content:
+      multipart/form-data:
+        schema:
+          type: object
+          properties:
+            dataset:
+              description: A list of updatable dataset properties
+              type: object
+              properties:
+                name:
+                  type: string
+                  description: The dataset name
+                datasetName:
+                  type: string
+                  deprecated: true
+                  description: Legacy value for backwards compatibility, same as the dataset name
+                dataSetId:
+                  type: integer
+                  format: int32
+                  description: The dataset id, integer format
```

Comment:
Sorry I missed this before, but this isn't an updatable field in `DatasetUpdate`. We get the ID from the path parameter.